__tests__/login/oidc/TokenRequester.spec.ts

```diff
@@ -0,0 +1,244 @@
+    }
+  };
+  function setUpMockedReturnValues(
+    values: Partial<typeof defaultReturnValues>
+  ): void {
+    defaultMocks.storageUtility.getForUser.mockResolvedValueOnce(
+      values.storageIdp ?? defaultReturnValues.storageIdp
+    );
+    defaultMocks.storageUtility.getForUser.mockResolvedValueOnce(
+      values.storageClientId ?? defaultReturnValues.storageClientId
+    );
+    defaultMocks.storageUtility.getForUser.mockResolvedValueOnce(
+      values.storageClientSecret ?? defaultReturnValues.storageClientSecret
+    );
+    const issuerConfig =
+      values.issuerConfig ?? defaultReturnValues.issuerConfig;
+    defaultMocks.issueConfigFetcher.fetchConfig.mockResolvedValueOnce(
+      issuerConfig
+    );
+    defaultMocks.fetcher.fetch.mockResolvedValueOnce(
+      (new NodeResponse(
+        values.responseBody ?? defaultReturnValues.responseBody
+      ) as unknown) as Response
+    );
+    defaultMocks.joseUtility.decodeJWT.mockReset();
+    defaultMocks.joseUtility.decodeJWT.mockResolvedValueOnce(
+      values.jwt ?? defaultReturnValues.jwt
+    );
+  }
+
+  it("Properly follows the refresh flow", async () => {
+    setUpMockedReturnValues({});
+    const TokenRefresher = getTokenRequester();
+    /* eslint-disable @typescript-eslint/camelcase */
+    await TokenRefresher.request("global", {
+      grant_type: "refresh_token",
+      refresh_token: "thisIsARefreshToken"
+    });
+    /* eslint-enable @typescript-eslint/camelcase */
+    expect(defaultMocks.fetcher.fetch).toBeCalledWith(
+      IssuerConfigFetcherFetchConfigResponse.tokenEndpoint,
+      {
+        method: "POST",
+        headers: {
+          DPoP: DpopHeaderCreatorResponse,
+          "content-type": "application/x-www-form-urlencoded"
+        },
+        body:
+          "grant_type=refresh_token&refresh_token=thisIsARefreshToken&client_id=coolApp"
+      }
+    );
+  });
+
+  it("Adds an authorization header if a client secret is present", async () => {
+    setUpMockedReturnValues({
+      storageClientSecret: "theSecretSauceIsJustMayo"
+    });
+    const TokenRefresher = getTokenRequester();
+    /* eslint-disable @typescript-eslint/camelcase */
+    await TokenRefresher.request("global", {
+      grant_type: "refresh_token",
+      refresh_token: "thisIsARefreshToken"
+    });
+    /* eslint-enable @typescript-eslint/camelcase */
+    expect(defaultMocks.fetcher.fetch).toBeCalledWith(
+      IssuerConfigFetcherFetchConfigResponse.tokenEndpoint,
+      {
+        method: "POST",
+        headers: {
+          DPoP: DpopHeaderCreatorResponse,
+          Authorization: "Basic Y29vbEFwcDp0aGVTZWNyZXRTYXVjZUlzSnVzdE1heW8=",
+          "content-type": "application/x-www-form-urlencoded"
+        },
+        body:
+          "grant_type=refresh_token&refresh_token=thisIsARefreshToken&client_id=coolApp"
+      }
+    );
+  });
+
+  it("Fails elegantly if the idp returns a bad value", async () => {
```

Comment:
Multiple "it("Fails elegantly if..." references in these tests, which doesn't really make sense (i.e. all failures should be 'elegant'!) - so should just be replaced with 'standard' wording like "it("should fail if...".